docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
+- Horizontally scalable (capacity and performance)
+- Accessible from a central end point
+- Highly Available
+- Easy to use query language
+- Service accessible through a UI
+- Monitorable service (health, performances and alerts)
+- Ability to correlate logs with KPIs and alerts
+- Multitenancy for audit logs
+- Performances (TBD)
+
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  system error investigation
+
+- As a Platform Administrator, I want to browse all system logs (from all
+  servers) from a unique endpoint, in order to ease distributed system error
+  investigation
+
+- As a Storage Administrator, I want to browse all Hyperdrive containers logs
+  (from all servers) from a unique endpoint, in order to ease Hyperdrive error
+  investigation
+
+- As a Storage Administrator, I want to browse all Zenko containers logs (from
+  all servers) from a unique endpoint, in order to ease Zenko error
+  investigation
+
+- As a Storage Administrator, I want to log all actions performed by users
+  (whatever their Role) on the RING X product, in order to be able to sustain
+  an audit
```

Comment:
We can't put this as a requirement on the log collection system: Loki/ES/anything doesn't care about what the logs are made of, it's the responsibility of the sender to define. So, if you want audit logs to cover all the product, maybe worth another epic.

Also, this will extend the requirement a bit: we said container logs (so, `stdout`/`stderr`) and system logs (to be defined, but probably journald entries). Now, if we consider audit logs (e.g. from kube-apiserver), then we also need to tail arbitrary files written by containers (not just the standard streams).

Reply:
Ok I removed everything about audit logs. Again, those US are the ones from the RINGX service map, just to give a little bit of context on what we need at a higher level